lib/src/component_declaration/component_base.dart

```diff
@@ -195,16 +195,19 @@ abstract class UiComponent<TProps extends UiProps> extends react.Component imple
     validateRequiredProps(appliedProps);
   }
 
+  /// Validates that props with the `@requiredProp` annotation are present.
   void validateRequiredProps(Map appliedProps) {
-    consumedProps?.forEach((ConsumedProps consumedProps) {
-      consumedProps.props.forEach((PropDescriptor prop) {
-        if (!prop.isRequired) return;
-        if (prop.isNullable && appliedProps.containsKey(prop.key)) return;
-        if (!prop.isNullable && appliedProps[prop.key] != null) return;
-
-        throw new PropError.required(prop.key, prop.errorMessage);
+    if (inReactDevMode) {
```

Comment:
Why do we have a check in this method but not in `validateProps`? Seems like it should be both or neither.

I lean toward neither, to allow consumers to still explicitly perform validation if desired when using the prod bundle, but that validation being turned off "by default" due to the checks in lifecycle methods.